url3ed4973106_kosmasK_RecordPlayer_tgz-pJ8-es_uc3m_recordplayer_test_TestJ8/src/es/uc3m/eda/list/singlelink/SList.java

```diff
@@ -0,0 +1,236 @@
+package es.uc3m.eda.list.singlelink;
+
+import es.uc3m.eda.list.IList;
+import es.uc3m.eda.list.IListIterator;
+
+public class SList<E> implements IList<E>, ISList<E>  {
+
+	SNode<E> firstNode = null;
+
+	@Override
+	public void addFirst(E newElem) {
+		SNode<E> newNode = new SNode<E>(newElem);
+		newNode.nextNode = this.firstNode;
+		this.firstNode = newNode;
+	}
+
+	@Override
+	public void addLast(E newElem) {
+		SNode<E> node = new SNode<E>(newElem);
+		SNode<E> lastNode = null;
+		for (SNode<E> nodeIt = this.firstNode; nodeIt != null; nodeIt = nodeIt.nextNode) {
+			lastNode = nodeIt;
+		}
+		if (lastNode == null) {
+			this.firstNode = node;
+		} else {
+			lastNode.nextNode = node;
+		}
+	}
+
+	@Override
+	public void insertAt(int index, E newElem) {
+		SNode<E> newNode = new SNode<E>(newElem);
+		if (index == 0) {
+			newNode.nextNode = this.firstNode;
+			this.firstNode = newNode;
+		} else {
+			int i = 1;
+			for (SNode<E> nodeIt = this.firstNode; nodeIt != null; nodeIt = nodeIt.nextNode) {
+				if (i == index) {
+					newNode.nextNode = nodeIt.nextNode;
+					nodeIt.nextNode = newNode;
+					return;
+				}
+				++i;
+			}
+			System.out.println("SList: Insertion out of bounds");
+		}
+	}
+
+	@Override
+	public boolean isEmpty() {
+		return (this.firstNode == null);
+	}
+
+	@Override
+	public boolean contains(E elem) {
+		for (SNode<E> nodeIt = this.firstNode; nodeIt != null; nodeIt = nodeIt.nextNode) {
+			if (nodeIt.getElement().equals(elem)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public void removeFirst() {
+		if (this.firstNode != null) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of != on line 68 to == check before setting the field value on line 69 or verify if the check is needed at all.